### PR TITLE
Refactor some GameParser code.

### PR DESCRIPTION
**Pull request by asvitkine**
## Change Summary & Additional Notes
Refactor some GameParser code.

Moves various helpers for dealing with variables to GameDataVariableParser and adds test coverage for them. Cleans up some related code.

These changes were the pure refactoring / clean up changes extracted from https://github.com/triplea-game/triplea/pull/10425.

### CODE REVIEW

File: game-app/game-core/src/main/java/games/strategy/engine/data/gameparser/GameParser.java [hunk 1]
```diff
@@ -182,6 +179,7 @@ private GameData parse(final Path xmlFile, final InputStream stream)
 
     parseDelegates(game.getGamePlay().getDelegates());
 
+    variableParser.parseVariables(game.getVariableList());
```
Comment (DanVanAtta) [inline game-app/game-core/src/main/java/games/strategy/engine/data/gameparser/GameParser.java:182]: There is a side-effect in this call where `parseVariables` caches the variables that were parsed.

WDYT if we made this more functional and `variableParser` returned those parsed variables and did not have any side effects?

**Review by DanVanAtta — COMMENTED** [review]
Just one comment regarding concern about side-effects

**Comment by DanVanAtta** [comment]
@asvitkine unrelated question: are you on forums? If so - what's your username there (DM LaFayette on forums).

File: game-app/game-core/src/main/java/games/strategy/engine/data/gameparser/GameParser.java (hunk 1 above)
Comment (asvitkine) [reply game-app/game-core/src/main/java/games/strategy/engine/data/gameparser/GameParser.java:182]: That's how it was before, see line 206 in the previous diff, but then GameParser had all these functions to operate on these variables. This change moves all that functionality to GameDataVariableParser, so it's all together.

We could refactor things so there's two classes, so there's a wrapper over the returned map. Or another option is we rename GameDataVariableParser to GameDataVariables and have a static parse() function that returns it, this way we don't need two classes.

**Comment by asvitkine** [comment]
> @asvitkine unrelated question: are you on forums? If so - what's your username there (DM LaFayette on forums).

Messaged you on the forums.

File: game-app/game-core/src/main/java/games/strategy/engine/data/gameparser/GameParser.java (hunk 1 above)
Comment (asvitkine) [reply game-app/game-core/src/main/java/games/strategy/engine/data/gameparser/GameParser.java]: OK, I ended up doing my proposed refactoring above. Please take another look.

**Review by DanVanAtta — APPROVED** [review]

**Pull request merged**